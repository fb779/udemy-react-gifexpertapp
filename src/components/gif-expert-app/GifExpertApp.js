import React, {useState} from 'react';
import GifAddCategory from './GifAddCategory';
import GifGrid from './GifGrid';

const GifExpertApp = () => {
  const [categories, setcategories] = useState(['Samurai X']);

  return (
    <>
      <GifAddCategory setcategories={setcategories} />
      <hr />
      <div>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </div>
    </>
  );
};

export default GifExpertApp;
