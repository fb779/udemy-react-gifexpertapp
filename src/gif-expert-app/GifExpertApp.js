import React, {useState} from 'react';
import PropTypes from 'prop-types';
import GifAddCategory from './GifAddCategory';
import GifGrid from './GifGrid';

const GifExpertApp = ({defaultCategories = []}) => {
  const [categories, setcategories] = useState(defaultCategories);

  return (
    <>
      <GifAddCategory setcategories={setcategories} />
      <hr />
      <div>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </div>
    </>
  );
};

GifExpertApp.propTypes = {
  defaultCategories: PropTypes.array,
};

export default GifExpertApp;
