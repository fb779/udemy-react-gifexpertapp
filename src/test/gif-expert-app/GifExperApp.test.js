import React from 'react';
import {shallow} from 'enzyme';
import GifExpertApp from '../../gif-expert-app/GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
  // let wrapper = shallow(<GifExpertApp />);
  // beforeEach(() => {
  //   wrapper = shallow(<GifExpertApp />);
  // });

  test('Prueba de carga del componente ', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe cargar una lista de categorias especificas ', () => {
    const categories = ['ip man', 'saitama'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
