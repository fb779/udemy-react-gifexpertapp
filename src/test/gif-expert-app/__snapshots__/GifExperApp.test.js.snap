// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe cargar una lista de categorias especificas  1`] = `
<Fragment>
  <GifAddCategory
    setcategories={[Function]}
  />
  <hr />
  <div>
    <GifGrid
      category="ip man"
      key="ip man"
    />
    <GifGrid
      category="saitama"
      key="saitama"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Prueba de carga del componente  1`] = `
<Fragment>
  <GifAddCategory
    setcategories={[Function]}
  />
  <hr />
  <div />
</Fragment>
`;
